### ISSUE TITLE
Remove web2 from trunk

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4821]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#4821|
|Type|enhancement|
|Created|2011-01-24 23:11:55Z|
|Branch|https://github.com/twisted/twisted/tree/remove-web2-4821-3|

Web2 isn't being maintained any more, and confuses people, so we should probably remove it from the repository. To make it easy to find the code if we want to reuse parts of it (as we did for http headers), a tag of trunk from right before the deletion may be helpful.

References to web2 will also probably need to be removed from the packaging code.

<details><summary>Searchable metadata</summary>

```
trac-id__4821 4821
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_remove_web2_4821_3 branches/remove-web2-4821-3
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1295910715000000 1295910715000000
changetime__1299818102000000 1299818102000000
version__None None
owner__exarkun exarkun
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

I suppose if we do this, we can also close all of the other open web2 tickets (or redirect them against web, as appropriate).

Anyone want to speak up against doing this?  Otherwise I think we should, and sooner rather than later.

The one objection I've heard so far is that removing it will make it hard to get the code.  I think we can address this by doing a tag before the delete.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30703]) Branching to 'remove-web2-4821'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_1):
> Anyone want to speak up against doing this?  Otherwise I think we should, and sooner rather than later.
Closed [#3870](https://github.com/twisted/twisted/issues/3870).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [30872]) Branching to 'remove-web2-4821-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Okay, I removed it.  There was a dependency on web2 in vfs, so I deleted some vfs functionality and disabled the unit test for it.  Since vfs isn't part of what we release and no one appears to be working on or using vfs, this doesn't really bother me.  The functionality wouldn't be that hard to re-instate based on the old producer/consumer interfaces, if someone really wanted to.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set owner to @exarkun|
|-|-|

I can't tell if this is a trick, but here's some feedback in the hopes that it will help web2 die:

 * the StreamProducer and IByteStream imports were removed from ftp.py in 30707 but are still used in FTPReadVFS:
```
twisted/vfs/adapters/ftp.py:270: undefined name 'StreamProducer'
twisted/vfs/adapters/ftp.py:270: undefined name 'IByteStream'
```

 * There are a couple of lingering references to web2:
```
./twisted/python/zsh/_mktap:"web2:An HTTP/1.1 web server that can serve from a filesystem or application resource."
./twisted/python/zsh/_mktap:web2)
./twisted/python/zsh/_mktap:'--class=[A class that will be used to serve the root resource.  Must implement twisted.web2.iweb.IResource and take no arguments.]:class:_files' \
./twisted/python/zsh/_twistd:"web2:An HTTP/1.1 web server that can serve from a filesystem or application resource."
./twisted/python/zsh/_twistd:web2)
./twisted/python/zsh/_twistd:'--class=[A class that will be used to serve the root resource.  Must implement twisted.web2.iweb.IResource and take no arguments.]:class:_files' \
./twisted/web/_auth/__init__.py:HTTP header-based authentication migrated from web2
./twisted/web/test/test_cgi.py:    filters = sys.executable,  # web2's version
```

 * If there's not already a ticket for this, the website, including TwistedWebPlan, will need to get updated.

Other than that, [the build looks fine](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-web2-4821-2).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [30898]) Remove web2 zsh completion information (manually, untested)

refs [#4821](https://github.com/twisted/twisted/issues/4821)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [30927]) Branching to 'remove-web2-4821-3'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

(In [30930]) Similar zsh completion info removal; untested, manual, like the _mktap change

refs [#4821](https://github.com/twisted/twisted/issues/4821)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [30931]) Get rid of the thing for web2's version

refs [#4821](https://github.com/twisted/twisted/issues/4821)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun removed owner|
|-|-|

Thanks for the review!  It is sort of difficult to decide if this ticket is a trick or not, I'm not entirely sure myself. :)

  1. Addressed vfs with [#4934](https://github.com/twisted/twisted/issues/4934) (sort of cheating, but effective)
  1. Got rid of almost all those references, but I did leave behind the one in the web auth package, since it's still true.
  1. Good idea!  I'll just go make some edits now, actually.

Here are the [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-web2-4821-3)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @glyph<br>@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_15"></a>|@glyph set owner to @exarkun<br>@glyph set status to `new`|
|-|-|

CONGRATURATION, A WINNER IS YOU.

` 94 files changed, 12 insertions(+), 20256 deletions(-)`

(Land.)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun set status to `closed`|
|-|-|

(In [30932]) Merge remove-web2-4821-3 

Author: exarkun
Reviewer: jesstess, glyph
Fixes: [#4821](https://github.com/twisted/twisted/issues/4821)

Remove twisted.web2, unreleased, unmaintained, incomplete.
